cli/CMakeLists.txt

```diff
@@ -30,6 +30,12 @@ if(LIBNUMA_FOUND)
   set(QUICKSTEP_HAVE_LIBNUMA TRUE)
 endif()
 
+if (BUILD_SHARED_LIBS)
+  set(GFLAGS_LIB_NAME gflags_nothreads-shared)
+else()
+  set(GFLAGS_LIB_NAME gflags_nothreads-static)
+endif()
```

Comment:
Try remove this part, as I believe `cmake` is smart enough to figure out the variable `GFLAGS_LIB_NAME`.

Otherwise, try use `options(GFLAGS_LIB_NAME ...)`.